lib/View.php

```diff
@@ -57,4 +57,22 @@ public function draw($template)
         extract($this->_variables);
         include $path;
     }
+
+    /**
+     * echo script tag incl. SRI hash for given script file
+     *
+     * @access private
+     * @param  string $file
+     * @param  bool   $async should it execute ASAP or only after HTML got parsed
+     */
+    private function _scriptTag($file, $async = true)
+    {
+        $sri = array_key_exists($file, $this->_variables['SRI']) ?
+            ' integrity="' . $this->_variables['SRI'][$file] . '"' : '';
+        $suffix = preg_match('#\d.js$#', $file) == 0 ?
+            '?' . rawurlencode($this->_variables['VERSION']) : '';
+        echo '<script ', $async ? 'async' : 'defer',
```

Comment:
Is it again my PHP knowledge lacking or why do you return the string comma-separated and not concatenated with `.`?

Note the PHPdoc here also does not document the return value, so I'm a little puzzled…

Reply:
There is no return value, the string is directly echo-ed. Since this is a private method, we can't (easily) unit test it, so I felt no need to add the extra return and add echos before every call to it in the templates.

This then allows a further micro-optimization: When using an echo keyword (like return, this is not a function) we can avoid the creation of an extra string in memory to catenate all the pieces into, but just pass all pieces as arguments to echo, hence the commas. See some other echos used in the templates, for example:
https://github.com/PrivateBin/PrivateBin/blob/3cba170f3255de21bbebb77f6c565519ef33e8c1/tpl/bootstrap5.php#L165